runtime/utils/common/src/aws/smithy/kotlin/runtime/util/Attributes.kt

```diff
@@ -123,4 +123,12 @@ private class AttributesImpl : Attributes {
 
     override val keys: Set<AttributeKey<*>>
         get() = map.keys
+
+    override fun equals(other: Any?): Boolean =
```

Comment:
same question, why define equality for this? What does it mean to have equality on a property bag of arbitrary values (that may not have meaningful equality defined).